stl/athena/killing11l/src/task_2/thread_pool.h

```diff
@@ -0,0 +1,45 @@
+#ifndef THREAD_POOL_H
+#define THREAD_POOL_H
+
+#include <atomic>
+#include <functional>
+#include <future>
+#include <iostream>
+#include <queue>
+#include <stdexcept>
+#include <thread>
+
+class ThreadPool {
+ public:
+  ThreadPool() = delete;
+  explicit ThreadPool(size_t);
+
+  template <class FUN, class... ARGS>
+  auto PushTask(FUN&& f, ARGS&&... args)
+      -> std::future<typename std::result_of<FUN(ARGS...)>::type> {
+    using ret_type = typename std::result_of<FUN(ARGS...)>::type;
+    auto task = std::make_shared<std::packaged_task<ret_type()>>(
+        std::bind(std::forward<FUN>(f), std::forward<ARGS>(args)...));
+
+    auto res = task->get_future();
+    {
+      std::unique_lock<std::mutex> lock(mtx_);
+      tasks_.emplace([task] { (*task)(); });
+    }
+    cond_.notify_all();
```

Comment:
may be use notify_one

Reply:
Ok, so notify_one is really better for this scenario.